plugins/woocommerce/includes/wc-order-functions.php

```diff
@@ -162,10 +162,27 @@ function wc_get_is_pending_statuses() {
  * @return string
  */
 function wc_get_order_status_name( $status ) {
-	$statuses = wc_get_order_statuses();
-	$status   = OrderUtil::remove_status_prefix( $status );
+	// "Special statuses": these are in common usage across WooCommerce, but are not normally returned by
+	// wc_get_order_statuses(). We add them only to avoid unnecessary doing-it-wrong errors.
+	$special_statuses = array(
+		'wc-' . OrderStatus::AUTO_DRAFT => OrderStatus::AUTO_DRAFT,
+		'wc-' . OrderStatus::TRASH      => OrderStatus::TRASH,
+	);
+
+	// Merge order is important. If the special statuses are ever returned by wc_get_order_statuses(), those definitions
+	// should take priority.
+	$statuses   = array_merge( $special_statuses, wc_get_order_statuses() );
+	$unprefixed = OrderUtil::remove_status_prefix( (string) $status );
```

Comment:
Just thinking out loud. Casting it explicitly to (string) should avoid fatal due to the wrong type being passed, though will not produce anything meanful if let's say an Object or Array is passed. But that's okay, since we don't want to restrict users from what they can or can't pass.

Throwing a doing it wrong, should do the job for creating awareness .

Reply:
So in fact, if an object is passed in, we **do** still get a fatal error. From the PR description:

> There are a few cases where a fatal error still happens, such as if an object is supplied instead of a string.

Though even that's a simplification, as there are exotic possibilities like stringifiable objects, which would be handled just fine. In any case, generally we'd get a fatal. My opinion is that's perfectly reasonable here, and even desirable.

The concrete reports we've had around this problem all related to `null` values, and that is handled nicely, imo. If an object is passed in, that's a different kettle of fish, and I don't think we should try to handle it near-silently (but it's also not a situation we know of in practice).